### DISCUSSION TITLE
Ability to add number of files changed or number of directories changed to the PR title

### DISCUSSION THREAD

**Discussion by nitrocode** [renovatebot/renovate#31579] [Suggest an Idea]
### Tell us more.

I have a PR title like this for a terraform upgrade of the aws provider

```
chore(deps): update terraform aws to v5
```

I'd like it to also include the number of directories updated so I can visually see how large of a change this is before jumping into the PR.

It would be nice if it said something like

```
chore(deps): update terraform aws to v5 - 12 dirs
```

Is this possible? if not, is this something that could be added?

Other options

- number of files affected
- number of directories affected

**Comment by secustor** [comment]
Take a look at the [`upgrades` template variable](https://docs.renovatebot.com/templates/) and handlebars native functions. 
That should give you everything you need.

**Reply by nitrocode** [reply]
Thank you. Ill give this a try

```json5
  prTitle: "{{ commitMessage }} ({{ upgrades.length }})"
```

How do i know how the object looks like to collect the files from each upgrade in order to count them to add to the suffix

**Reply by secustor** [reply]
You can see those as in the branch status debug logs.

**Reply by nitrocode** [reply]
I can use `upgrades.length` to get the number of upgrades, but getting the number of unique directories is harder or impossible using mustache.

I'd need to use a mustache template to iterate through the list of objects called `upgrades`, look at each `packageFile` key, figure out which directory the file is in, and return the number of unique directories. 

This would be easier if there was a `packageDirectory` key exposed. 

Or is there another way to do this?

**Reply by secustor** [reply]
A single `packageDirectory` will not be possible as packages can be grouped independently of where they are located. 

What would be possible is to add some more helpers for such use cases.
If you can come some handlebar helpers we can consider them.
